draft-packages/button/KaizenDraft/Button/styles.scss

```diff
@@ -0,0 +1,467 @@
+      calc(#{$ca-grid * 0.5} - #{$kz-border-focus-ring-border-width});
+  }
+
+  :global(.js-focus-visible) %caButton%caButtonForm:global(.focus-visible) & {
+    padding: calc(#{$ca-grid / 3} - #{$kz-border-focus-ring-border-width})
+      calc(#{$ca-grid * 1} - #{$kz-border-focus-ring-border-width});
+  }
+
+  :global(.js-focus-visible)
+    %caButtonSecondary%caButtonForm:global(.focus-visible)
+    & {
+    padding: calc(#{$ca-grid / 3} - #{$kz-border-focus-ring-border-width})
+      calc(#{$ca-grid * 0.5} - #{$kz-border-focus-ring-border-width});
+  }
+}
+
+%caButtonLabel {
+  // This line height equates to 3/4 of a grid unit - so 18px. We set this
+  // here rather than in ca-type-ideal-button because changing the type style
+  // changes the baseline grid alignment for all buttons. Changing it here keeps
+  // the baseline alignment correct for single line buttons.
+  line-height: 1;
+
+  &:only-child {
+    %caButtonSecondary & {
+      margin: 0;
+    }
+  }
+
+  &:not(:first-child) {
+    @include ca-margin($start: 0.5em);
+  }
+
+  &:not(:last-child) {
+    @include ca-margin($end: 0.5em);
+  }
+}
+
+%caButtonIconWrapper {
+  height: $caButtonIcon-height;
+  align-self: flex-start;
+}
+
+%caButtonIconButton {
+  width: $caButton-height;
+
+  // Reset for various states and variants
+  &:not(:disabled),
+  &:disabled {
+    &,
+    &:hover,
+    &:active,
+    &%caButtonPrimary,
+    &%caButtonSecondary {
+      color: $kz-color-wisteria-800;
+      background: none;
+      border-color: $kz-border-borderless-border-color;
+
+      &%caButtonReversed {
+        color: white;
+        &%caButtonPrimary {
+          %caButtonContent {
+            color: white;
+            background: none;
+          }
+        }
+      }
+    }
+  }
+  &%caButtonReversed,
+  :global(.js-focus-visible) &:global(.focus-visible) {
+    %caButtonContent {
+      padding: 0;
+    }
+  }
+
+  %caButtonContent {
+    justify-content: center;
+    width: 100%;
+    padding: 0;
+    // IE 10/11 workaround to fix vertical centering of icon
+    margin: calc(
+        (((#{$caButton-height}) - #{$caButtonIcon-height}) / 2) - #{$kz-border-solid-border-width}
+      )
+      0;
+  }
+
+  &%caButtonForm {
+    width: $caButton-formHeight;
+  }
+
+  &%caButtonDestructive:not(:disabled) {
+    &:hover,
+    &:active {
+      color: $kz-color-coral-600;
+    }
+
+    :global(.js-focus-visible) &:global(.focus-visible) {
+      border-color: $kz-color-cluny-500;
+    }
+  }
+}
+
+%caButtonDestructiveModifier {
+  &:not(:disabled) {
+    color: $kz-color-coral-600;
+
+    %caButtonLabel {
+      color: $kz-color-coral-600;
+    }
+  }
+
+  &:hover span,
+  &:active span {
+    color: $kz-color-coral-600;
+  }
+
+  &:hover,
+  &:active,
+  &:global(.focus-visible) {
+    // These are only used by consumers, so the !important is there to make sure these
+    // override any local competing styles that we can't foresee
+    color: $kz-color-coral-600 !important;
```

Comment:
Should this have :global(.js-focus-visible) as well? - to follow the pattern of the other user .focus-visible usages (I guess it's trying to restrict that it's trying to restrict it so that it only applies if the plugin is actually used). I'm having a hard time understanding "These are only used by consumers" - is this something that should be deprecated? Is it due to a "reacher" somewhere? How did you discover this? Is there anything special about this rule such that "to make sure these override any local competing styles that we can't foresee" only applies here and not everywhere else? I'm sure there's a reason, I'm just having a hard time following it.

Reply:
Fair, it is a pretty crappy comment. I'll document stuff here as well and maybe you can tell me how much of this you think should go in the comment:

Here's a link to [the current Button version of this on master, for comparison](https://github.com/cultureamp/kaizen-design-system/blob/c04eb2440e4166e699afe8d84ae10b30ab438e9e/packages/component-library/components/Button/styles.scss#L430).

> is this something that should be deprecated?

No. It's currently being used only by the Elm button – someone added it in the past as a hack to create a Button variant that was both `secondary` and `destructive`. There is no secondary destructive Button in React yet. As for the Elm one, there is only 1 usage of it in Murmur:

![image](https://user-images.githubusercontent.com/6406263/83103087-544f3780-a0f9-11ea-857d-b879fd6ae06b.png)

There does exist a Zen design for a Secondary Destructive Button, but since A) it's not implemented in React yet and thus not used anywhere, and B) we want to redesign the API anyway, I thought I'd leave the React implementation for later.

![image](https://user-images.githubusercontent.com/6406263/83103175-8a8cb700-a0f9-11ea-8dc2-cbaaa8b52e57.png)

> Is it due to a "reacher" somewhere?

Sort of. My memory is vague but I remember when I tried to apply the focus ring code to this variant, it was losing the specificity battle due to something that was particular to the Elm component and so I assumed there might be reachers out there that were depending on things like the reached-into `span` (see lines above the comment), so I didn't want to remove it.

To be honest I don't have a clear understanding of why this hack was done the way it was, and once I realised that the Elm implementation was quite different to the React one, this seemed like the safer choice at the time to make sure any reachers we might not know about didn't win out. Now I can't even look at the Elm story in Storybook to see what this variant looks like, because this PR hasn't been merged yet which means the Elm package isn't published yet, so I had to remove the Elm story for it to work. I can follow up with a second PR after this to bring the Elm story back and check it.

> Is there anything special about this rule such that "to make sure these override any local competing styles that we can't foresee" only applies here and not everywhere else?

Just that `%caButtonDestructiveModifier` only exists in the hack used solely in the Elm implementation of the Button.

Reply:
Ah, this does bring back some memories now. It might worth just mentioning that it's only used in the Elm version and that the Elm version of the Button has a slightly different Html structure (as far as I understand) - as a warning if anyone desperately needs to make a React destructive button before the API gets rewritten.

Yes, good chance we'll want to completely rewrite that. There so many scss mixins and overrides going on, it's quite hard to follow - hopefully there's a simpler way.

Reply:
Roger, I'll reword it to include those details 👍 Cheers